hydroflow/src/compiled/pull/symmetric_hash_join.rs

```diff
@@ -34,17 +38,29 @@ where
     }
 
     fn build(&mut self, k: Key, v: &ValBuild) -> bool {
-        let vec = self.table.entry(k).or_insert_with(Vec::new);
-        if !vec.contains(v) {
-            vec.push(v.clone());
-            return true;
-        }
+        let entry = self.table.entry(k);
+
+        match entry {
+            Entry::Occupied(mut e) => {
+                let vec = e.get_mut();
+
+                if !vec.contains(v) {
+                    vec.push(v.clone());
+                    return true;
+                }
+            }
+            Entry::Vacant(e) => {
+                e.insert(vec![v.clone()]);
+                return true;
+            }
+        };
+
         false
     }
 
     fn probe(&mut self, k: Key, v: ValProbe) {
-        if self.table.contains_key(&k) {
-            self.current_matches = Some((k, v, 0));
+        if let Some(entry) = self.table.get(&k) {
+            self.current_matches = Some((k, v, 0, entry as *const _));
```

Comment:
Option 1: instead of keeping a raw pointer to the vec, steal ownership of the vec via `self.table.remove(&k)` and replace it after all matches are emitted

Reply:
Option 2: Just do all the cloning and pairing up here. Something like `entry.clone().into_iter().map(|v| (k.clone(), v)).collect::<Vec<_>>()` and then do (the equivalent of a) `flatten` afterwards